allennlp/commands/evaluate.py

```diff
@@ -101,21 +101,35 @@ def evaluate(model: Model,
         logger.info("Iterating over dataset")
         generator_tqdm = Tqdm.tqdm(iterator, total=data_iterator.get_num_batches(instances))
 
+        # Number of batches in instances.
         batch_count = 0
+        # Number of batches where the model produces a loss.
         loss_count = 0
+        # Cumulative weighted loss
         total_loss = 0.0
+        # Cumulative weight across all batches.
+        total_weight = 0.0
 
         for batch in generator_tqdm:
             batch_count += 1
             batch = util.move_to_device(batch, cuda_device)
-            loss = model(**batch).get("loss")
+            output_dict = model(**batch)
+            loss = output_dict.get("loss")
 
             metrics = model.get_metrics()
 
             if loss is not None:
                 loss_count += 1
-                metrics["loss"] = loss.item()
-                total_loss += loss.item()
+                weight = output_dict.get("batch_weight")
```

Comment:
I'm not a fan of magic constants like this ("loss" is kind of a necessary evil).

How would you feel about having it be a command line arg `batch_weight_key` or something. As a bonus, that would give us a place to *document* this feature, otherwise no one will even know about it unless they read the code.